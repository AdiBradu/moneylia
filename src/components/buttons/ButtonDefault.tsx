import React from 'react';
import {TouchableOpacity, Text} from 'react-native';
//Styling
import styled from 'styled-components';
//Types
import {ButtonDefaultProps} from '../../types/types';

const StyledButton = styled(TouchableOpacity)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 16px 20px;
  width: 260px;
  height: 50px;
  background: #7476ed;
  border-radius: 4px;
  margin-top: 60px;
`;

const StyledLabel = styled(Text)`
  font-family: 'WorkSans-Medium';
  font-size: 18px;
  line-height: 18px;
  color: #ffffff;
`;

export const ButtonDefault: React.FC<ButtonDefaultProps> = ({
  text,
  onPress,
  ...props
}) => {
  return (
    <StyledButton onPress={onPress} {...props}>
      <StyledLabel>{text}</StyledLabel>
    </StyledButton>
  );
};
