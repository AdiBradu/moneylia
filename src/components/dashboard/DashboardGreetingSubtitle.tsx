import React from 'react';
import {Text} from 'react-native';
//Styling
import styled from 'styled-components';
import {theme} from '../../styles/theme';

const StyledText = styled(Text)`
  font-family: 'WorkSans-Regular';
  font-size: 14px;
  line-height: 21px;
  letter-spacing: -0.5px;
  color: ${theme.colors.white};
`;

export const DashboardGreetingSubtitle = () => {
  return <StyledText>Welcome in Moneylia</StyledText>;
};
