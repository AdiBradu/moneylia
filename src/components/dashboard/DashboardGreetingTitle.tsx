import React from 'react';
import {Text} from 'react-native';
//Styling
import styled from 'styled-components';
import {theme} from '../../styles/theme';

const Styledtext = styled(Text)`
  font-family: 'WorkSans-Regular';
  font-size: 32px;
  line-height: 38.4px;
  color: ${theme.colors.white};
`;

export const DashboardGreetingTitle = () => {
  return <Styledtext>Hello, John S.</Styledtext>;
};
