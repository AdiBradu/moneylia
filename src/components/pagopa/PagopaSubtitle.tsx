import React from 'react';
import {View, Text} from 'react-native';
//Styling
import {theme} from '../../styles/theme';
import styled from 'styled-components';

const StyledText = styled(Text)`
  font-family: 'Rubik-Regular';
  font-size: 14px;
  line-height: 21px;
  color: ${theme.colors.copy};
  margin-top: 8px;
  margin-bottom: 12px;
`;

export const PagopaSubtitle: React.FC = () => {
  return (
    <View>
      <StyledText>
        View pending transactions for the past two years and proceed to payment
      </StyledText>
    </View>
  );
};
