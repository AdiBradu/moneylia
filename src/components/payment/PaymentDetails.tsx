import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
//Styling
import {theme} from '../../styles/theme';
import styled from 'styled-components';
//Assets
import {ChevronLeftIcon} from '../../assets/icons/ChevronLeftIcon';

interface PaymentDetailsPops {
  onPress: () => void;
}

const StyledView = styled(View)`
  width: 100%;
  height: 60px;
  display: flex;
  flex-direction: row;
  background-color: ${theme.colors.pagopa};
  align-items: center;
  justify-content: center;
`;

const StyledText = styled(Text)`
  font-family: 'WorkSans-Regular';
  font-size: 16px;
  line-height: 19.2px;
  color: ${theme.colors.white};
`;

const StyledChevron = styled(TouchableOpacity)<PaymentDetailsPops>`
  position: absolute;
  left: 16px;
  cursor: pointer;
`;

export const PaymentDetails: React.FC<PaymentDetailsPops> = ({onPress}) => {
  return (
    <StyledView>
      <StyledChevron onPress={onPress}>
        <ChevronLeftIcon />
      </StyledChevron>
      <StyledText>Payment details</StyledText>
    </StyledView>
  );
};
