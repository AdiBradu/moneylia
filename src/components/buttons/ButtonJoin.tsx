import React from 'react';
//Styling
import styled from 'styled-components';
import {theme} from '../../styles/theme';
//Components
import {ButtonDefault} from './ButtonDefault';
//Navigation
import {useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack/lib/typescript/src/types';
//Types
import {RootStackParamList} from '../../types/types';

const StyledButton = styled(ButtonDefault)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 16px 20px;
  width: 260px;
  height: 50px;
  background: ${theme.colors.primary};
  border-radius: 4px;
  margin-top: 60px;
`;

export const ButtonJoin: React.FC = () => {
  const navigation =
    useNavigation<NativeStackNavigationProp<RootStackParamList>>();
  const handlePress = () => {
    navigation.navigate('Tab');
  };

  return <StyledButton onPress={handlePress} text="Join for free" />;
};
