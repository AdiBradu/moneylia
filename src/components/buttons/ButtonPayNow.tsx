import React from 'react';
//Styling
import styled from 'styled-components';
import {theme} from '../../styles/theme';
//Components
import {ButtonDefault} from './ButtonDefault';

const StyledButton = styled(ButtonDefault)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 16px 20px;
  width: 100%;
  height: 50px;
  background: ${theme.colors.pagopa};
  border-radius: 4px;
  margin-top: 16px;
`;

export const ButtonPayNow: React.FC = () => {
  const handlePress = () => {
    console.log('Pay now');
  };

  return <StyledButton onPress={handlePress} text="Pay now" />;
};
