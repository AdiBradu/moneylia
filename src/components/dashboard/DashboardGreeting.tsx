import React from 'react';
import {View} from 'react-native';
//Components
import {DashboardGreetingTitle} from './DashboardGreetingTitle';
import {DashboardGreetingSubtitle} from './DashboardGreetingSubtitle';
//Styling
import styled from 'styled-components';
import {theme} from '../../styles/theme';

const StyledView = styled(View)`
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const DashboardGreeting = () => {
  return (
    <StyledView>
      <DashboardGreetingTitle />
      <DashboardGreetingSubtitle />
    </StyledView>
  );
};
