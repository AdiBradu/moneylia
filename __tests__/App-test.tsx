/**
 * @format
 */

import 'react-native';
import React from 'react';

import {ButtonPayNow} from '../src/components/buttons/ButtonPayNow';
import {ButtonDefault} from '../src/components/buttons/ButtonDefault';
import {DashboardGreeting} from '../src/components/dashboard/DashboardGreeting';
import {DashboardGreetingTitle} from '../src/components/dashboard/DashboardGreetingTitle';
import {DashboardGreetingSubtitle} from '../src/components/dashboard/DashboardGreetingSubtitle';
import {PagoPAIcon} from '../src/assets/icons/PagoPAIcon';
import {PagopaSubtitle} from '../src/components/pagopa/PagopaSubtitle';
import {MoneyliaLogomark} from '../src/assets/icons/MoneyliaLogomark';
import {PaymentDetails} from '../src/components/payment/PaymentDetails';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('ButtonPayNow renders correctly', () => {
  const tree = renderer.create(<ButtonPayNow />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ButtonJoin renders correctly', () => {
  const tree = renderer
    .create(
      <ButtonDefault
        text={'Press Me'}
        onPress={() => console.log('Button pressed')}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('DashboardGreetingTitle renders correctly', () => {
  const tree = renderer.create(<DashboardGreetingTitle />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('DashboardGreetingSubtitle renders correctly', () => {
  const tree = renderer.create(<DashboardGreetingSubtitle />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('DashboardGreeting renders correctly', () => {
  const tree = renderer.create(<DashboardGreeting />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('PagoPAIcon renders correctly', () => {
  const tree = renderer.create(<PagoPAIcon color={'#fff'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('PagopaSubtitle renders correctly', () => {
  const tree = renderer.create(<PagopaSubtitle />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('MoneyliaLogomark renders correctly', () => {
  const tree = renderer.create(<MoneyliaLogomark color={'#7476ED'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('PaymentDetails renders correctly', () => {
  const tree = renderer
    .create(<PaymentDetails onPress={() => console.log('Go Back!')} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
