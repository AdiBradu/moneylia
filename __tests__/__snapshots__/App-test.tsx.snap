// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonJoin renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    className="sc-gswNZR lkmccW"
  >
    Press Me
  </Text>
</View>
`;

exports[`ButtonPayNow renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    className="sc-gswNZR lkmccW"
  >
    Pay now
  </Text>
</View>
`;

exports[`DashboardGreeting renders correctly 1`] = `
<View
  className="sc-jSUZER cRTBLW"
>
  <Text
    className="sc-hLBbgP daVEtE"
  >
    Hello, John S.
  </Text>
  <Text
    className="sc-eDvSVe hpOgkx"
  >
    Welcome in Moneylia
  </Text>
</View>
`;

exports[`DashboardGreetingSubtitle renders correctly 1`] = `
<Text
  className="sc-eDvSVe hpOgkx"
>
  Welcome in Moneylia
</Text>
`;

exports[`DashboardGreetingTitle renders correctly 1`] = `
<Text
  className="sc-hLBbgP daVEtE"
>
  Hello, John S.
</Text>
`;

exports[`MoneyliaLogomark renders correctly 1`] = `
<View>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="28"
    bbWidth="28"
    fill="none"
    focusable={false}
    height="28"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 28,
          "width": 28,
        },
      ]
    }
    vbHeight={28}
    vbWidth={28}
    width="28"
  >
    <RNSVGGroup
      fill={null}
      propList={
        Array [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M8.69464 22.6737C6.28938 25.0451 2.21579 25.0357 1.07815 21.8751C0.855416 21.2563 0.677332 20.6201 0.546351 19.9709C0.0115605 17.3201 0.286035 14.5726 1.33507 12.0756C2.3841 9.57867 4.16056 7.4445 6.43983 5.94297C8.71912 4.44145 11.3988 3.64001 14.1401 3.64001C16.8813 3.64001 19.561 4.44145 21.8402 5.94297C24.1195 7.4445 25.896 9.57867 26.945 12.0757C27.994 14.5726 28.2685 17.3201 27.7337 19.9709C27.6028 20.6201 27.4247 21.2563 27.2019 21.8751C26.0642 25.0357 21.9907 25.0451 19.5854 22.6737L15.8725 19.0131C15.5178 18.6634 15.5545 18.0866 15.6523 17.6015C15.7118 17.3067 15.6812 17.001 15.5645 16.7232C15.4478 16.4454 15.2502 16.208 14.9967 16.041C14.7431 15.874 14.445 15.7848 14.1401 15.7848C13.8351 15.7848 13.537 15.874 13.2834 16.041C13.0298 16.208 12.8322 16.4454 12.7155 16.7232C12.5988 17.001 12.5683 17.3067 12.6278 17.6015C12.7257 18.0866 12.7622 18.6634 12.4075 19.0131L8.69464 22.6737Z"
        fill={
          Object {
            "payload": 4285822701,
            "type": 0,
          }
        }
        propList={
          Array [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`PagoPAIcon renders correctly 1`] = `
<View>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="28"
    bbWidth="29"
    fill="none"
    focusable={false}
    height="28"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 28,
          "width": 29,
        },
      ]
    }
    vbHeight={28}
    vbWidth={29}
    width="29"
  >
    <RNSVGGroup
      fill={null}
      propList={
        Array [
          "fill",
        ]
      }
    >
      <RNSVGPath
        clipRule={0}
        d="M19.9518 12.7573L19.9591 10.6372H19.9664V10.5787H21.3107C22.2044 10.5787 22.6477 10.3374 22.6477 9.84031V6.34579C22.6404 5.84866 22.1754 5.6001 21.2598 5.6001H20.1408H19.9664H18.4114V12.7573H19.9518ZM19.9446 6.42621H21.0999V9.74525H19.9446V6.42621ZM3.49338 14.8847L3.50064 12.7646V12.7061H4.62694C5.37538 12.7061 5.74598 12.4649 5.74598 11.9677V8.47322C5.7387 7.97609 5.35358 7.72753 4.58334 7.72753H3.64597H3.50064H2.19995V14.8847H3.49338ZM3.49338 8.56095H4.45981V11.88H3.49338V8.56095ZM9.72812 12.7061H7.27206C6.55268 12.7061 6.18934 12.4576 6.18934 11.9604L6.19662 10.5056C6.19662 10.0158 6.5672 9.76721 7.31564 9.75989H8.44194V8.56095H7.47552L7.46824 9.2189H6.2039V8.47322C6.2039 7.9834 6.58176 7.74215 7.33744 7.74215L8.60182 7.72753C9.35026 7.72753 9.72812 7.97609 9.72812 8.46591V12.7061ZM7.48276 11.8727H8.4565L8.44922 10.5933H7.48276V11.8727ZM11.4209 12.7061H12.5545L12.5617 12.7646V14.0659H11.5953V13.4079H10.3092V14.1463C10.3092 14.6434 10.687 14.892 11.4355 14.892H12.7216C13.47 14.892 13.8479 14.6434 13.8479 14.1463V7.72753H12.5617H12.4019H11.4645C10.6943 7.72753 10.3092 7.97609 10.3019 8.47322V11.9677C10.3019 12.4576 10.6725 12.7061 11.4209 12.7061ZM12.5545 11.8727H11.588V8.56095H12.5545V11.8727ZM17.8298 11.9604C17.8226 12.4576 17.4375 12.6988 16.6672 12.6988H15.4174C14.6617 12.6988 14.2838 12.4502 14.2838 11.9604V8.46591C14.2838 7.96878 14.6617 7.72753 15.4174 7.72753H16.7035C17.452 7.72753 17.8226 7.97609 17.8226 8.46591V11.9604H17.8298ZM16.5509 8.56095H15.5772V11.88H16.5437L16.5509 8.56095ZM26.0554 5.64385C26.9492 5.64385 27.3924 5.8851 27.3924 6.38223L27.3997 12.3989C27.3997 17.9185 22.9526 22.4 17.4592 22.4C14.2765 22.4 11.4354 20.894 9.61871 18.5545L8.63048 19.3441L8.49241 14.9211L11.5879 17.0193L10.8177 17.6115C12.3582 19.5927 14.7634 20.8721 17.4592 20.8721C22.1097 20.8721 25.8665 17.0924 25.8665 12.4136V10.5859H24.7039V12.7718H23.1634V6.38223C23.1634 5.89241 23.6139 5.64385 24.5149 5.64385H26.0554ZM24.6966 9.75245H25.8592V6.46265H24.6966V9.75245Z"
        fill={
          Object {
            "payload": 4294967295,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          Array [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`PagopaSubtitle renders correctly 1`] = `
<View>
  <Text
    className="sc-gKPRtg hGvKMl"
  >
    View pending transactions for the past two years and proceed to payment
  </Text>
</View>
`;

exports[`PaymentDetails renders correctly 1`] = `
<View
  className="sc-iBYQkv ibCXFV"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="28"
        bbWidth="28"
        fill="none"
        focusable={false}
        height="28"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 28,
              "width": 28,
            },
          ]
        }
        vbHeight={28}
        vbWidth={28}
        width="28"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            clipRule={0}
            d="M18.325 5.00846C18.7807 5.46407 18.7807 6.20276 18.325 6.65837L10.9833 14.0001L18.325 21.3418C18.7807 21.7974 18.7807 22.5361 18.325 22.9917C17.8694 23.4473 17.1307 23.4473 16.6751 22.9917L8.50846 14.825C8.05285 14.3694 8.05285 13.6307 8.50846 13.1751L16.6751 5.00846C17.1307 4.55285 17.8694 4.55285 18.325 5.00846Z"
            fill={
              Object {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              Array [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <Text
    className="sc-ftTHYK bbpqGw"
  >
    Payment details
  </Text>
</View>
`;
